hack/update-spelling.sh

```diff
@@ -0,0 +1,43 @@
+#!/usr/bin/env bash
+# Copyright 2023 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+TOOL_VERSION="v0.3.4"
+
+ROOT_DIR="$(realpath "$(dirname "${BASH_SOURCE[0]}")"/..)"
+allowd_spelling_words="${ROOT_DIR}/hack/allowd_spelling_words"
+
+COMMAND=()
+if command -v misspell; then
+  COMMAND=(misspell)
+elif command -v "${ROOT_DIR}/bin/misspell"; then
+  COMMAND=("${ROOT_DIR}/bin/misspell")
+else
+  #NOTE we use go install refer the go site :https://go.dev/doc/go-get-install-deprecation
+  GOBIN="${ROOT_DIR}/bin/" go install github.com/client9/misspell/cmd/misspell@${TOOL_VERSION}
+  COMMAND=("${ROOT_DIR}/bin/misspell")
+fi
+
+function check() {
+  # check spelling
+  echo "Checking spelling..."
+  ignore="$(tr <"${allowd_spelling_words}" '\n' ',')"
+  "${COMMAND[@]}" -i "${ignore}" "$(find . -not \( -path ./vendor\* -o -path ./.git\* \))"
```

Comment:
```suggestion
  ignore="$(tr <"${allowd_spelling_words}" '\n' ',')"
  "${COMMAND[@]}" -w -i "${ignore}" "$(find . -not \( -path ./vendor\* -o -path ./.git\* \))"
```

Reply:
Good suggestion.
with -w flag , So misspell can override the file.
And then ```git --no-pager diff --exit-code``` can check the diff of file.

Update later.